src/mscorlib/src/System/Span.cs

```diff
@@ -612,7 +612,13 @@ internal static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
         {
             if (byteLength == 0)
                 return;
-
+            
+#if AMD64
+            if (byteLength > uint.MaxValue) goto PInvoke;
```

Comment:
Can this be something smaller than uint.MaxValue, like several kB - similar to what is done for memcpy?

The implementation of InitBlockUnaligned is not friendly to the GC. It can lead to long GC pauses for large sizes. The PInvoke path does not have this problem.